src/Metadata/Extractor/XmlPropertyExtractor.php

```diff
@@ -0,0 +1,142 @@
+                'readable' => $this->phpize($property, 'readable', 'bool'),
+                'writable' => $this->phpize($property, 'writable', 'bool'),
+                'readableLink' => $this->phpize($property, 'readableLink', 'bool'),
+                'writableLink' => $this->phpize($property, 'writableLink', 'bool'),
+                'required' => $this->phpize($property, 'required', 'bool'),
+                'identifier' => $this->phpize($property, 'identifier', 'bool'),
+                'default' => $this->phpize($property, 'default', 'string'),
+                'example' => $this->phpize($property, 'example', 'string'),
+                'deprecationReason' => $this->phpize($property, 'deprecationReason', 'string'),
+                'fetchable' => $this->phpize($property, 'fetchable', 'bool'),
+                'fetchEager' => $this->phpize($property, 'fetchEager', 'bool'),
+                'jsonldContext' => isset($property->jsonldContext->values) ? $this->buildValues($property->jsonldContext->values) : null,
+                'openapiContext' => isset($property->openapiContext->values) ? $this->buildValues($property->openapiContext->values) : null,
+                'push' => $this->phpize($property, 'push', 'bool'),
+                'security' => $this->phpize($property, 'security', 'string'),
+                'securityPostDenormalize' => $this->phpize($property, 'securityPostDenormalize', 'string'),
+                'types' => $this->buildArrayValue($property, 'type'),
+                'builtinTypes' => $this->buildArrayValue($property, 'builtinType'),
+                'schema' => isset($property->schema->values) ? $this->buildValues($property->schema->values) : null,
+                'initializable' => $this->phpize($property, 'initializable', 'bool'),
+                'extraProperties' => $this->buildExtraProperties($property, 'extraProperties'),
+            ];
+        }
+    }
+
+    private function buildExtraProperties(\SimpleXMLElement $resource, string $key = null): ?array
+    {
+        if (null !== $key) {
+            if (!isset($resource->{$key})) {
+                return null;
+            }
+
+            $resource = $resource->{$key};
+        }
+
+        return $this->buildValues($resource->values);
+    }
+
+    /**
+     * @return string[]
+     */
+    private function buildValues(\SimpleXMLElement $resource): array
+    {
+        $data = [];
+        foreach ($resource->value as $value) {
+            if (null !== $value->attributes()->name) {
+                $data[(string) $value->attributes()->name] = isset($value->values) ? $this->buildValues($value->values) : (string) $value;
+                continue;
+            }
+
+            $data[] = isset($value->values) ? $this->buildValues($value->values) : (string) $value;
+        }
+
+        return $data;
+    }
+
+    private function buildArrayValue(?\SimpleXMLElement $resource, string $key, $default = null)
```

Comment:
This method, as well as `phpize` and maybe others are used into XmlPropertyExtractor and XmlResourceExtractor should we use a trait with these to reduce the code base?